lib/stats/stats-cluster.c

```diff
@@ -39,21 +39,34 @@ stats_cluster_foreach_counter(StatsCluster *self, StatsForeachCounterFunc func,
     }
 }
 
+static const gchar *tag_names[SC_TYPE_MAX] =
+{
+  /* [SC_TYPE_DROPPED]   = */ "dropped",
+  /* [SC_TYPE_PROCESSED] = */ "processed",
+  /* [SC_TYPE_STORED]   = */  "stored",
+  /* [SC_TYPE_SUPPRESSED] = */ "suppressed",
+  /* [SC_TYPE_STAMP] = */ "stamp",
+};
+
 const gchar *
 stats_cluster_get_type_name(gint type)
 {
-  static const gchar *tag_names[SC_TYPE_MAX] =
-  {
-    /* [SC_TYPE_DROPPED]   = */ "dropped",
-    /* [SC_TYPE_PROCESSED] = */ "processed",
-    /* [SC_TYPE_STORED]   = */  "stored",
-    /* [SC_TYPE_SUPPRESSED] = */ "suppressed",
-    /* [SC_TYPE_STAMP] = */ "stamp",
-  };
-
   return tag_names[type];
 }
 
+gint
+stats_cluster_get_type_by_name(const gchar *name)
+{
+  if (!name)
+    return SC_TYPE_MAX;
+
```

Comment:
well.. that enum is a little bit strange.
`SC_TYPE_MIN` is 0, as well as `SC_TYPE_DROPPED`,
but `SC_TYPE_MAX` is not equal to `SC_TYPE_STAMP`.
Of course, we can use -1, -2, -n, but it still not symmetric.
So I think  `SC_TYPE_MIN` and `SC_TYPE_MAX` should be the extremal values here :)